zerver/lib/outgoing_webhook.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import absolute_import
+from typing import Any, Iterable, Dict, Tuple, Callable, Text, Mapping
+
+import requests
+import json
+import sys
+import inspect
+import logging
+from six.moves import urllib
+from functools import reduce
+
+from django.utils.translation import ugettext as _
+
+from zerver.models import Realm, get_realm_by_email_domain, get_user_profile_by_id
+from zerver.lib.actions import internal_send_message
+from zerver.lib.queue import queue_json_publish
+from zerver.lib.validator import check_dict, check_string
+from zerver.decorator import JsonableError
+
+class ServiceMessageActions(object):
```

Comment:
Should we get rid of this class and just have these functions be normal functions?  We're obviously not actually using the class for anything.

I feel like other interfaces we might add, are unlikely to swap these methods out; instead, they'll either (1) want to modify the shape of the payload in `do_rest_call` (e.g. "Slack format") or (2) want to not call `do_rest_call` at all, as in the case of `contrib_bots` using this API (they'll just call some other function).